genesyscloud/provider/provider.go

```diff
@@ -239,37 +210,112 @@ func GetRegionBasePath(region string) string {
 	return "https://api." + getRegionDomain(region)
 }
 
+func configure(version string) schema.ConfigureContextFunc {
+	return func(context context.Context, data *schema.ResourceData) (interface{}, diag.Diagnostics) {
+		// Initialize a single client if we have an access token
+		accessToken := data.Get("access_token").(string)
+		if accessToken != "" {
+			Once.Do(func() {
+				sdkConfig := platformclientv2.GetDefaultConfiguration()
+				_ = InitClientConfig(data, version, sdkConfig)
+
+				SdkClientPool = &SDKClientPool{
+					Pool: make(chan *platformclientv2.Configuration, 1),
+				}
+				SdkClientPool.Pool <- sdkConfig
+			})
+		} else {
+			// Initialize the SDK Client pool
+			err := InitSDKClientPool(data.Get("token_pool_size").(int), version, data)
+			if err != nil {
+				return nil, err
+			}
+		}
+		return &ProviderMeta{
+			Version:      version,
+			ClientConfig: platformclientv2.GetDefaultConfiguration(),
+			Domain:       getRegionDomain(data.Get("aws_region").(string)),
+		}, nil
+	}
+}
+
 func InitClientConfig(data *schema.ResourceData, version string, config *platformclientv2.Configuration) diag.Diagnostics {
```

Comment:
I didn't make any changes to this function I just pulled out some of its functionality into their own functions